core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournalConfig.scala

```diff
@@ -45,12 +45,35 @@ class CassandraJournalConfig(system: ActorSystem, config: Config) extends Cassan
     }(collection.breakOut)
   }
 
+  private val tags: HashMap[String, Int] = loadTagMap("tags")
+
+  private val useTagPrefixes: Boolean = config.getBoolean("use-tag-prefixes")
+
+  private val tagPrefixes: HashMap[String, Int] = loadTagMap("tag-prefixes")
+
+  def idForTag(tag: String): Int =
+    if (useTagPrefixes) {
+      tagPrefixes
+        .collectFirst {
+          case (prefix, index) if tag.startsWith(prefix) =>
+            index
+        }
+        .getOrElse(1)
+    } else {
+      tags.getOrElse(tag, 1)
+    }
```

Comment:
shouldn't it look for exact match first, and if no match look at prefixes?

Reply:
See my reply about flag.